authorization/src/cryptography.rs

```diff
@@ -16,7 +16,7 @@ fn check_ed25519_auth(e: &Env, auth: &KeyedEd25519Signature, function: Symbol, p
         function,
         contrct_id: e.get_current_contract(),
         network_id: e.ledger().network_passphrase(),
-        parameters: parameters.try_into().unwrap(),
+        parameters: parameters.to_raw().try_into_val(e).unwrap(),
```

Comment:
This isn't really ideal in terms of ergonomics. We can probably implement `TryIntoVal` on `EnvVal` for all types that implement it on `RawVal` and pass through so we don't need to do the `.to_raw()`.